db/dcrpg/pgblockchain_charts_test.go

```diff
@@ -72,26 +39,45 @@ func TestMain(m *testing.M) {
 // It also checks the order and duplicates in the x-axis dataset.
 func TestPgCharts(t *testing.T) {
 	charts := cache.NewChartData(context.Background(), 0, &chaincfg.MainNetParams)
-	charts.DiffInterval = 10
+
 	db.RegisterCharts(charts)
+	// Register a dummy updater for Fees, PoolSize, and PoolValue. This must be
+	// registered *after* ChainDB registers its updaters, which set the correct
+	// length of blocks.Time and the other slices.
+	registerDummyFeeAndPoolInfo(charts)
+
+	// Validator for the Blocks and Windows chart data slice lengths.
 	blocks := charts.Blocks
 	windows := charts.Windows
 
 	validate := func(tag string) {
 		// Not checking NewAtoms right now, as the test database does not appear to
 		// contain stakebase and coinbase vins.
-		_, err := cache.ValidateLengths(blocks.Time, blocks.Chainwork, blocks.TxCount, blocks.BlockSize)
+		_, err := cache.ValidateLengths(blocks.Time, blocks.Chainwork,
+			blocks.TxCount, blocks.BlockSize, blocks.Fees, blocks.PoolSize,
+			blocks.PoolValue)
```

Comment:
`blocks.NewAtoms` is missing.

Reply:
added

Reply:
@buck54321 You had a comment a couple lines up about omitting NewAtoms because it's not in the test DB.  Is that OK now given the data you regenerated a few days ago?  It works for me with a full DB, but I've not tried the test DBs we have in https://github.com/dcrlabs/bug-free-happiness/tree/master/pgdb